packages/expo-calendar/src/Calendar.ts

```diff
@@ -558,6 +558,157 @@ export type DaysOfTheWeek = {
 
 export { PermissionResponse, PermissionStatus, PermissionHookOptions };
 
+/**
+ * Enum containing all possible user responses to the calendar UI dialogs.
+ * */
+export enum CalendarDialogResultActions {
+  /**
+   * On Android, this is the only possible result because the OS doesn't provide enough information to determine the user's action -
+   * the user may have canceled the dialog, modified the event, or deleted it.
+   *
+   * On iOS, this means the user simply closed the dialog.
+   * */
+  done = 'done',
+  /**
+   * The user canceled or dismissed the dialog.
+   * @platform ios
+   * */
+  canceled = 'canceled',
+  /**
+   * The user deleted the event.
+   * @platform ios
+   * */
+  deleted = 'deleted',
+  /**
+   * The user responded to and saved a pending event invitation.
+   * @platform ios
+   * */
+  responded = 'responded',
+  /**
+   * The user modified an existing event.
+   * @platform ios
+   * */
+  saved = 'saved',
+}
+
+/**
+ * The result of presenting the calendar dialog for viewing an event. It indicates how user responded to the dialog.
+ * On Android, the `action` is always `done`. On iOS, more `action` values are possible.
+ * */
+export type ViewEventDialogResult = {
+  action: Extract<CalendarDialogResultActions, 'done' | 'canceled' | 'deleted' | 'responded'>;
```

Comment:
```suggestion
 * The result of presenting the calendar dialog for viewing an event. 
 */
export type ViewEventDialogResult = {
  /**
  * Indicates how user responded to the dialog.
  * On Android, it is always `'done'`. On iOS, more values are possible.
  */
  action: Extract<CalendarDialogResultActions, 'done' | 'canceled' | 'deleted' | 'responded'>;
```

We can move the part of description to the key.

Reply:
did that